### ISSUE TITLE
NoMethodError: undefined method `to_sym' for nil:NilClass on NetSuite::Records::Customer(customer_id, credentials)

### ISSUE THREAD

**Issue by adambedford** [NetSweet/netsuite#182]
I'm seeing this error when making a `get` request to Customer. I'm not too familiar with NetSuite and am using a developer sandbox. Looks like the error is related to custom fields, and I haven't yet touched custom fields in this developer sandbox.

Any thoughts?

```
NoMethodError: undefined method `to_sym' for nil:NilClass
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/records/custom_field_list.rb:17:in `block in initialize'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/records/custom_field_list.rb:15:in `each'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/records/custom_field_list.rb:15:in `initialize'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/support/fields.rb:34:in `new'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/support/fields.rb:34:in `block in field'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/support/attributes.rb:15:in `block in initialize_from_attributes_hash'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/support/attributes.rb:14:in `each'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/support/attributes.rb:14:in `initialize_from_attributes_hash'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/records/customer.rb:47:in `initialize'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/actions/get.rb:72:in `new'
	from /Users/Adam/.rvm/gems/ruby-2.1.2/bundler/gems/netsuite-35b3e07babab/lib/netsuite/actions/get.rb:72:in `get'
```

**Comment by iloveitaly** [comment]
Can you post the full XML response that you are getting from NetSuite?
Might be a custom field customization that isn't handled yet.

**Comment by adambedford** [comment]
```
2.1.2 :046 >   NetSuite::Records::Customer.get(1542, options)
I, [2015-05-12T14:13:21.032016 #49977]  INFO -- : SOAP request: https://webservices.na1.netsuite.com/services/NetSuitePort_2014_2
I, [2015-05-12T14:13:21.032189 #49977]  INFO -- : SOAPAction: "get", Content-Type: text/xml;charset=UTF-8, Content-Length: 845
D, [2015-05-12T14:13:21.032495 #49977] DEBUG -- : <?xml version="1.0" encoding="UTF-8"?>
<env:Envelope xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:platformMsgs="urn:messages_2014_2.platform.webservices.netsuite.com" xmlns:env="http://schemas.xmlsoap.org/soap/envelope/" xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com">
  <env:Header>
    <platformMsgs:passport>
      <platformCore:email>bedfo101@mail.chapman.edu</platformCore:email>
      <platformCore:password>Mavenlink1+</platformCore:password>
      <platformCore:account>TSTDRV1280464</platformCore:account>
      <platformCore:role internalId="3"/>
    </platformMsgs:passport>
  </env:Header>
  <env:Body>
    <platformMsgs:get>
      <platformMsgs:baseRef xsi:type="platformCore:RecordRef" internalId="1542" type="customer"/>
    </platformMsgs:get>
  </env:Body>
</env:Envelope>

D, [2015-05-12T14:13:21.032726 #49977] DEBUG -- : HTTPI POST request to webservices.na1.netsuite.com (net_http)
I, [2015-05-12T14:13:22.162704 #49977]  INFO -- : SOAP response (status 200)
D, [2015-05-12T14:13:22.164076 #49977] DEBUG -- : <?xml version="1.0" encoding="UTF-8"?>
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <soapenv:Header>
    <platformMsgs:documentInfo xmlns:platformMsgs="urn:messages_2014_2.platform.webservices.netsuite.com">
      <platformMsgs:nsId>WEBSERVICES_TSTDRV1280464_051220151241091800435838488_8c0a21</platformMsgs:nsId>
    </platformMsgs:documentInfo>
  </soapenv:Header>
  <soapenv:Body>
    <getResponse xmlns="urn:messages_2014_2.platform.webservices.netsuite.com">
      <readResponse>
        <platformCore:status xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" isSuccess="true"/>
        <record xmlns:listRel="urn:relationships_2014_2.lists.webservices.netsuite.com" internalId="1542" xsi:type="listRel:Customer">
          <listRel:customForm xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="16">
            <platformCore:name>Ramsey Default Customer Form</platformCore:name>
          </listRel:customForm>
          <listRel:entityId>Aaron Abbott</listRel:entityId>
          <listRel:isPerson>true</listRel:isPerson>
          <listRel:firstName>Aaron</listRel:firstName>
          <listRel:lastName>Abbott</listRel:lastName>
          <listRel:entityStatus xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="13">
            <platformCore:name>CUSTOMER-Closed Won</platformCore:name>
          </listRel:entityStatus>
          <listRel:phone>(303) 464-4122</listRel:phone>
          <listRel:email>aa@fake-email.com</listRel:email>
          <listRel:defaultAddress>Aaron Abbott
1500 Broadway
Denver CO 80202
United States</listRel:defaultAddress>
          <listRel:isInactive>false</listRel:isInactive>
          <listRel:language>_usEnglish</listRel:language>
          <listRel:dateCreated>2015-04-17T01:24:44.000-07:00</listRel:dateCreated>
          <listRel:subsidiary xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1">
            <platformCore:name>Honeycomb Mfg.</platformCore:name>
          </listRel:subsidiary>
          <listRel:salesRep xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1008">
            <platformCore:name>Mary Redding</platformCore:name>
          </listRel:salesRep>
          <listRel:territory xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="10">
            <platformCore:name>Western</platformCore:name>
          </listRel:territory>
          <listRel:partner xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="164">
            <platformCore:name>Jasper Supply</platformCore:name>
          </listRel:partner>
          <listRel:creditHoldOverride>_auto</listRel:creditHoldOverride>
          <listRel:balance>0.0</listRel:balance>
          <listRel:overdueBalance>0.0</listRel:overdueBalance>
          <listRel:unbilledOrders>945.9</listRel:unbilledOrders>
          <listRel:consolUnbilledOrders>945.9</listRel:consolUnbilledOrders>
          <listRel:consolOverdueBalance>0.0</listRel:consolOverdueBalance>
          <listRel:consolDepositBalance>0.0</listRel:consolDepositBalance>
          <listRel:consolBalance>0.0</listRel:consolBalance>
          <listRel:currency xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1">
            <platformCore:name>USA</platformCore:name>
          </listRel:currency>
          <listRel:depositBalance>0.0</listRel:depositBalance>
          <listRel:shipComplete>false</listRel:shipComplete>
          <listRel:taxable>true</listRel:taxable>
          <listRel:startDate>2015-04-17T00:00:00.000-07:00</listRel:startDate>
          <listRel:thirdPartyCountry>_unitedStates</listRel:thirdPartyCountry>
          <listRel:giveAccess>true</listRel:giveAccess>
          <listRel:accessRole xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="14">
            <platformCore:name>Customer Center</platformCore:name>
          </listRel:accessRole>
          <listRel:sendEmail>false</listRel:sendEmail>
          <listRel:receivablesAccount xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="-10">
            <platformCore:name>Use System Preference</platformCore:name>
          </listRel:receivablesAccount>
          <listRel:clickStream>Camcorders &gt; Home &gt; Cameras &gt; Cameras &gt; Shopping Cart</listRel:clickStream>
          <listRel:lastPageVisited>login-register</listRel:lastPageVisited>
          <listRel:visits>12</listRel:visits>
          <listRel:firstVisit>2015-04-17T01:22:16.000-07:00</listRel:firstVisit>
          <listRel:lastVisit>2015-05-02T11:54:44.000-07:00</listRel:lastVisit>
          <listRel:billPay>false</listRel:billPay>
          <listRel:lastModifiedDate>2015-04-08T23:08:02.000-07:00</listRel:lastModifiedDate>
          <listRel:stage>_customer</listRel:stage>
          <listRel:addressbookList>
            <listRel:addressbook>
              <listRel:defaultShipping>true</listRel:defaultShipping>
              <listRel:defaultBilling>true</listRel:defaultBilling>
              <listRel:isResidential>false</listRel:isResidential>
              <listRel:label>1500 Broadway</listRel:label>
              <listRel:addressbookAddress xmlns:platformCommon="urn:common_2014_2.platform.webservices.netsuite.com">
                <platformCommon:country>_unitedStates</platformCommon:country>
                <platformCommon:addressee>Aaron Abbott</platformCommon:addressee>
                <platformCommon:addrPhone>(303) 464-4122</platformCommon:addrPhone>
                <platformCommon:addr1>1500 Broadway</platformCommon:addr1>
                <platformCommon:city>Denver</platformCommon:city>
                <platformCommon:state>CO</platformCommon:state>
                <platformCommon:zip>80202</platformCommon:zip>
                <platformCommon:addrText>Aaron Abbott
1500 Broadway
Denver CO 80202
United States</platformCommon:addrText>
                <platformCommon:override>false</platformCommon:override>
              </listRel:addressbookAddress>
              <listRel:internalId>244955</listRel:internalId>
            </listRel:addressbook>
          </listRel:addressbookList>
          <listRel:subscriptionsList>
            <listRel:subscriptions>
              <listRel:subscribed>true</listRel:subscribed>
              <listRel:subscription xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="2">
                <platformCore:name>Billing Communication</platformCore:name>
              </listRel:subscription>
              <listRel:lastModifiedDate>2015-04-20T16:23:33.000-07:00</listRel:lastModifiedDate>
            </listRel:subscriptions>
            <listRel:subscriptions>
              <listRel:subscribed>true</listRel:subscribed>
              <listRel:subscription xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1">
                <platformCore:name>Marketing</platformCore:name>
              </listRel:subscription>
              <listRel:lastModifiedDate>2015-04-20T16:23:33.000-07:00</listRel:lastModifiedDate>
            </listRel:subscriptions>
            <listRel:subscriptions>
              <listRel:subscribed>true</listRel:subscribed>
              <listRel:subscription xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="4">
                <platformCore:name>Newsletters</platformCore:name>
              </listRel:subscription>
              <listRel:lastModifiedDate>2015-04-20T16:23:33.000-07:00</listRel:lastModifiedDate>
            </listRel:subscriptions>
            <listRel:subscriptions>
              <listRel:subscribed>true</listRel:subscribed>
              <listRel:subscription xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="5">
                <platformCore:name>Product Updates</platformCore:name>
              </listRel:subscription>
              <listRel:lastModifiedDate>2015-04-20T16:23:33.000-07:00</listRel:lastModifiedDate>
            </listRel:subscriptions>
            <listRel:subscriptions>
              <listRel:subscribed>true</listRel:subscribed>
              <listRel:subscription xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="3">
                <platformCore:name>Surveys</platformCore:name>
              </listRel:subscription>
              <listRel:lastModifiedDate>2015-04-20T16:23:33.000-07:00</listRel:lastModifiedDate>
            </listRel:subscriptions>
          </listRel:subscriptionsList>
          <listRel:contactRolesList>
            <listRel:contactRoles>
              <listRel:giveAccess>false</listRel:giveAccess>
              <listRel:contact xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1542">
                <platformCore:name>Aaron Abbott</platformCore:name>
              </listRel:contact>
              <listRel:email>aa@fake-email.com</listRel:email>
              <listRel:role xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="14">
                <platformCore:name>Customer Center</platformCore:name>
              </listRel:role>
              <listRel:sendEmail>false</listRel:sendEmail>
            </listRel:contactRoles>
          </listRel:contactRolesList>
          <listRel:currencyList>
            <listRel:currency>
              <listRel:currency xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="1">
                <platformCore:name>USA</platformCore:name>
              </listRel:currency>
              <listRel:balance>0.0</listRel:balance>
              <listRel:consolBalance>0.0</listRel:consolBalance>
              <listRel:depositBalance>0.0</listRel:depositBalance>
              <listRel:consolDepositBalance>0.0</listRel:consolDepositBalance>
              <listRel:overdueBalance>0.0</listRel:overdueBalance>
              <listRel:consolOverdueBalance>0.0</listRel:consolOverdueBalance>
              <listRel:unbilledOrders>945.9</listRel:unbilledOrders>
              <listRel:consolUnbilledOrders>945.9</listRel:consolUnbilledOrders>
              <listRel:overrideCurrencyFormat>false</listRel:overrideCurrencyFormat>
              <listRel:displaySymbol>$</listRel:displaySymbol>
              <listRel:symbolPlacement>_beforeNumber</listRel:symbolPlacement>
            </listRel:currency>
          </listRel:currencyList>
          <listRel:creditCardsList>
            <listRel:creditCards>
              <listRel:internalId>39</listRel:internalId>
              <listRel:ccNumber>************8562</listRel:ccNumber>
              <listRel:ccExpireDate>2019-03-01T00:00:00.000-08:00</listRel:ccExpireDate>
              <listRel:ccName>Aaron Abbott</listRel:ccName>
              <listRel:paymentMethod xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="5">
                <platformCore:name>Visa</platformCore:name>
              </listRel:paymentMethod>
              <listRel:cardState xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com" internalId="UNKNOWN">
                <platformCore:name>Unknown</platformCore:name>
              </listRel:cardState>
              <listRel:ccDefault>true</listRel:ccDefault>
            </listRel:creditCards>
          </listRel:creditCardsList>
          <listRel:customFieldList xmlns:platformCore="urn:core_2014_2.platform.webservices.netsuite.com">
            <platformCore:customField internalId="3993" scriptId="custentity_2663_direct_debit" xsi:type="platformCore:BooleanCustomFieldRef">
              <platformCore:value>false</platformCore:value>
            </platformCore:customField>
            <platformCore:customField internalId="4565" scriptId="custentity_cust_priority" xsi:type="platformCore:LongCustomFieldRef">
              <platformCore:value>50</platformCore:value>
            </platformCore:customField>
            <platformCore:customField internalId="201" scriptId="custentity_fmt_customer_credit_on_hold" xsi:type="platformCore:BooleanCustomFieldRef">
              <platformCore:value>false</platformCore:value>
            </platformCore:customField>
            <platformCore:customField internalId="183" scriptId="custentity_fmt_cust_credit_on_hold" xsi:type="platformCore:BooleanCustomFieldRef">
              <platformCore:value>false</platformCore:value>
            </platformCore:customField>
            <platformCore:customField xsi:type="platformCore:StringCustomFieldRef">
              <platformCore:value>&lt;script type='text/javascript' src='https://www.google.com/jsapi'&gt;&lt;/script&gt;&lt;script type='text/javascript'&gt;google.load('visualization', '1', {packages:['orgchart'], callback:drawChart});function drawChart() {var data = new google.visualization.DataTable();data.addColumn('string', 'Name');data.addColumn('string', 'Manager');data.addColumn('string', 'ToolTip');data.addRows([]);var chart = new google.visualization.OrgChart(document.getElementById('chart_div'));chart.draw(data, {allowHtml:true});}&lt;/script&gt;&lt;body&gt;&lt;div id='chart_div'&gt;&lt;/div&gt;&lt;/body&gt;</platformCore:value>
            </platformCore:customField>
          </listRel:customFieldList>
        </record>
      </readResponse>
    </getResponse>
  </soapenv:Body>
</soapenv:Envelope>
```

**Comment by iloveitaly** [comment]
You found a bug! Fixed this with f4c235a

Let me know if this doesn't work for you!
